packages/frontend/src/components/EditorLayout/index.tsx

```diff
@@ -136,7 +168,18 @@ export default function EditorLayout() {
           borderColor="base.divider.subtle"
         >
           <Flex flex={1} alignItems="center">
-            <Box as={Link} to={URLS.FLOWS} mt={1} mr={3}>
+            <Box
+              as={Link}
+              to={URLS.FLOWS}
+              mt={1}
+              mr={3}
+              onClick={(e: React.MouseEvent<HTMLAnchorElement>) => {
+                if (shouldWarnOnLeave) {
+                  e.preventDefault()
+                  onWarningOpen()
+                }
+              }}
+            >
```

Comment:
is it possible to add this for the settings icon too? but it should navigate to the `notifications` page instead